Make WebGPU crashtests more robust

heycam opened the pull request:
#### f96776e9a72272b6ecabdb0c7988fa65c1f4d9eb
<pre>
Make WebGPU crashtests more robust
<a href="https://bugs.webkit.org/show_bug.cgi?id=284007">https://bugs.webkit.org/show_bug.cgi?id=284007</a>
<a href="https://rdar.apple.com/140873189">rdar://140873189</a>

Reviewed by Mike Wyrzykowski.

Many of the shader validation crashtests are timing dependent. Use
`await device.queue.onSubmittedWorkDone()` at the end of the test to
make them more likely to crash before the test has completed.

Remove the finally() call so that unexpected exceptions cause a test
to fail.

Skip a subset of the tests on Intel Macs, which use features or limits
unsupported on that hardware.

* LayoutTests/fast/webgpu/nocrash/fuzz-126711484.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-128396311.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-272863.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-272903.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-272911.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273017.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273021.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273023.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273323.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273503.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273505.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273566.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273570.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273573.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273578.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274161.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274171.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274270.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274271.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274275.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274290.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274317.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274622.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275167.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275172.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275218.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275225.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275228.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275229.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275232.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275371.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-276279.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-277017.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-277557.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-277642.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-277864.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-277928.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-277928b.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-278049.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-278474.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-279086.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-279449.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-279453.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-279912.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-281271.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-281272.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-281539.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-281555.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-281603.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-281614.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-281778.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282052.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282086.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282097.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282116.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282485.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282499.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282710.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282995.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-283008.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-283051.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-283071.html:
* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/287587@main">https://commits.webkit.org/287587@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4b099c23b8624932b83bca720ba2f514b9686e65

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/80164 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/59166 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/33577 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/84681 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/31140 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/68227 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/7465 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/62666 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/20486 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/83233 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/52739 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/73012 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/42972 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/50069 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/27166 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/29600 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/71198 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/27673 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/86113 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/7384 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/5229 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/70942 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/7559 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/68851 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/70180 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/14177 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/13128 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12406 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/7348 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/7187 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/10707 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/8992 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski commented on LayoutTests/fast/webgpu/nocrash/fuzz-272863.html:
```diff
@@ -6,7 +6,9 @@
     let commandEncoder = device.createCommandEncoder();
     let querySet = device.createQuerySet({ type: 'occlusion', count: 52 });
     commandEncoder.resolveQuerySet(querySet, 3, 0, buffer, 11264)
+    await Promise.any([device.queue.onSubmittedWorkDone(), new Promise(r => setTimeout(r, 10000))]);
```
timeout shouldn't be needed here, taking a look

heycam committed:
Make WebGPU crashtests more robust
https://bugs.webkit.org/show_bug.cgi?id=284007
rdar://140873189

Reviewed by Mike Wyrzykowski.

Many of the shader validation crashtests are timing dependent. Use
`await device.queue.onSubmittedWorkDone()` at the end of the test to
make them more likely to crash before the test has completed.

Remove the finally() call so that unexpected exceptions cause a test
to fail.

Skip a subset of the tests on Intel Macs, which use features or limits
unsupported on that hardware.

* LayoutTests/fast/webgpu/nocrash/fuzz-126711484.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-128396311.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-272863.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-272903.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-272911.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273017.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273021.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273023.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273323.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273503.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273505.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273566.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273570.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273573.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-273578.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274161.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274171.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274270.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274271.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274275.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274290.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274317.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-274622.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275167.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275172.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275218.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275225.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275228.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275229.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275232.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-275371.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-276279.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-277017.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-277557.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-277642.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-277864.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-277928.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-277928b.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-278049.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-278474.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-279086.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-279449.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-279453.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-279912.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-281271.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-281272.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-281539.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-281555.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-281603.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-281614.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-281778.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282052.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282086.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282097.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282116.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282485.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282499.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282710.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-282995.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-283008.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-283051.html:
* LayoutTests/fast/webgpu/nocrash/fuzz-283071.html:
* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: https://commits.webkit.org/287587@main
